cmd/release-tools/github.go

```diff
@@ -0,0 +1,200 @@
+// Copyright (c) 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/blang/semver"
+	"github.com/google/go-github/github"
+	"golang.org/x/oauth2"
+	"golang.org/x/sys/unix"
+)
+
+type GitHubClient struct {
+	owner string
+	github.Client
+}
+
+func newGitHubClient(owner string, token string) (*GitHubClient, error) {
+	ctx := context.Background()
+
+	var tc *http.Client
+
+	if token != "" {
```

Comment:
if the user does not set a github token, the function will return `&GitHubClient{owner, *github.NewClient(tc)}, nil` anyway?

Reply:
It may be good, it is possible to use the github api without token, but is limited to 60 request per day.  At the moment I only check if the token is empty only doing `release` subcommand, for `status` .
So I have to options:
- Always ask for a token (even if is possible to use 60 request for `status) 
- Keep optional the token for `release-tool status <repo_name>` but document it in the README.

I'd go for the second, what do you think?

Reply:
possibly to show a warning when you do not have more available requests